source/world/vs_chunk_manager.cpp

```diff
@@ -207,6 +214,22 @@ void VSChunkManager::updateChunks()
     assert(debug_isMainThread());
 
     initializeChunks();
+
+    // Init from file asynchronous
+    bool expected = true;
+    if (bShouldInitializeFromData.compare_exchange_weak(expected, false))
+    {
+        // Set block data to chunks, TODO: add checks, this is potentially dangerous
+        uint32_t chunkSize = this->getChunkBlockCount();
+        auto iter = worldDataFromFile.blocks.begin();
+        for (const auto chunk : this->chunks)
+        {
+            chunk->blocks = std::vector<VSBlockID>(iter, iter + chunkSize);
```

Comment:
chunk.blocks is already pre allocated to contain chunkSize blocks, to avoid the additional copy use:
```suggestion
            chunk->blocks.insert(chunk->blocks.begin(), iter, iter + chunkSize);
```